src/ui/utils/sources.test.ts

```diff
@@ -94,27 +94,27 @@ describe("newSourcesToCompleteSourceDetails", () => {
       ])
     ).toEqual({
       "1": {
-        canonicalId: "1",
         contentHash: "contentHash#1",
         correspondingSourceIds: ["1"],
         generated: [],
-        generatedFrom: [],
+        generatedFrom: ["pp1"],
```

Comment:
Personally, I really don't like this. `generatedFrom` should mean that we have a sourcemap linking source `pp1` with the generated source `1`, and that is how we computed the content of `pp1`. Generated and original sources only make sense in the context of source-mapping, and that is not what pretty-printing is (or, if it was, it would be correct to call the pretty-printed file the `generated`, not the `original`). I'm not going to block on it, because I am not doing much with this code these days, but I find this a major source of confusion. I think if we continue to completely reverse the meanings of these terms in some contexts we will also continue to completely confuse the daylight out of every single person who ever tries to understand the code.

Reply:
We have 3 different reasons for original <-> generated links between sources: sourcemaps, pretty-printing and inline script extraction. We could store these 3 types of links separately but in our code we mostly use these links to determine which source is preferred for display - separating the links would make these parts of our code (slightly) more complicated while I can't think of a single place where it would help make the code simpler.
If we want to avoid confusion I think we should use different terms for these links but not make working with them in the code harder.